src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs

```diff
@@ -107,17 +107,17 @@ public static IMethodSymbol CreateEqualsMethod(this Compilation compilation, Imm
         ImmutableArray<ISymbol> members,
         string localNameOpt)
     {
-        using var _1 = ArrayBuilder<SyntaxNode>.GetInstance(out var statements);
 
         // A ref like type can not be boxed. Because of this an overloaded Equals taking object in the general case
         // can never be true, because an equivalent object can never be boxed into the object itself. Therefore only
         // need to return false.
         if (containingType.IsRefLikeType)
         {
-            statements.Add(factory.ReturnStatement(factory.FalseLiteralExpression()));
-            return statements.ToImmutable();
+            return [factory.ReturnStatement(factory.FalseLiteralExpression())];
         }
 
+        using var statements = TemporaryArray<SyntaxNode>.Empty;
```

Comment:
we should talk to sam about that, and do in another PR :)

imo, i'm wary abotu it.  because maybe you have a case that is 1 elements 99% of the time, but very rarely you go up to >4.  TempArray is great for that purpose.

Reply:
I guess my thought is that if I see a loop in the code, TemporaryArray is essentially not a good choice. I feel like those cases can just ArrayBuilder or FixedSizeArrayBuilder and work almost as well. I just feel like TemporaryArray is too easy to move off the happy path without realizing it, and if I have to give up that case you mentioned to not have to worry about that, I'm ok with that.

Reply:
there is no loop here.  this is 'CreateEqualsMethod' which is either 1, 2, or 3 statemnets depending on the form we're ccreating.

Reply:
> I hesitated making suggestions to switch to TemporaryArray in non-trivial codepaths, 

i agree with that.  i only switched to TempArray where i thought it was trivial.  :)

Reply:
I think our thoughts on this code being trivial differ :) .